### PR TITLE
Add better error messages to ml_api

### PR DESCRIPTION
**Pull request by nvtkaszpir**
```text
21:50:49 kaszpir@lynx ~/Downloads $ curl -v "http://127.0.0.1:13333/p/?img=http://bagno.hldspl/obico/bad_1.jpg"
*   Trying 127.0.0.1:13333...
* Connected to 127.0.0.1 (127.0.0.1) port 13333 (#0)
> GET /p/?img=http://bagno.hldspl/obico/bad_1.jpg HTTP/1.1
> Host: 127.0.0.1:13333
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 503 SERVICE UNAVAILABLE
< Server: gunicorn
< Date: Fri, 16 Feb 2024 20:51:09 GMT
< Connection: keep-alive
< Content-Type: application/json
< Content-Length: 399
< 
{
  "detections": [],
  "message": "Failed to get image ImmutableMultiDict([('img', 'http://bagno.hldspl/obico/bad_1.jpg')]) - HTTPConnectionPool(host='bagno.hldspl', port=80): Max retries exceeded with url: /obico/bad_1.jpg (Caused by NameResolutionError(\"<urllib3.connection.HTTPConnection object at 0x7fa26ae25760>: Failed to resolve 'bagno.hldspl' ([Errno -2] Name or service not known)\"))"
}
* Connection #0 to host 127.0.0.1 left intact
```

gunicorn log:
```text
[2024-02-16 20:51:09,449] ERROR in server: Failed to get image ImmutableMultiDict([('img', 'http://bagno.hldspl/obico/bad_1.jpg')]) - HTTPConnectionPool(host='bagno.hldspl', port=80): Max retries exceeded with url: /obico/bad_1.jpg (Caused by NameResolutionError("<urllib3.connection.HTTPConnection object at 0x7fa26ae25760>: Failed to resolve 'bagno.hldspl' ([Errno -2] Name or service not known)"))
172.17.0.1 - - [16/Feb/2024:20:51:09 +0000] "GET /p/?img=http://bagno.hldspl/obico/bad_1.jpg HTTP/1.1" 503 399 "-" "curl/7.81.0"
```

### CODE REVIEW

File: ml_api/server.py [hunk 1]
```diff
@@ -43,13 +43,29 @@ def get_p():
...
+        except Exception as err:
             sentry_sdk.capture_exception()
+            app.logger.error(f"Failed to get image {request.args} - {err}")
+            abort(
+                make_response(
+                    jsonify(
+                        detections=[],
+                        message=f"Failed to get image {request.args} - {err}",
+                    ),
+                    503,
```
Comment (kennethjiang) [inline ml_api/server.py:55]: Not that it matters that much. But 5xx usually means a server error. Some load balancer/health checker may act accordingly. Let's just use a 400 code here.

File: ml_api/server.py [hunk 2]
```diff
@@ -43,13 +43,29 @@ def get_p():
...
+            )
     else:
-        app.logger.warn("Invalid request params: {}".format(request.args))
+        app.logger.warn(f"Invalid request params: {request.args}")
+        abort(
+            make_response(
+                jsonify(
+                    detections=[], message=f"Invalid request params: {request.args}"
+                ),
+                400,
```
Comment (kennethjiang) [inline ml_api/server.py:65]: I'll be fine with 400. But the convention is probably 422. Again not that it matters that much.

File: ml_api/server.py (hunk 1 above)
Comment (nvtkaszpir) [reply ml_api/server.py]: updated error code to 400.

File: ml_api/server.py (hunk 2 above)
Comment (nvtkaszpir) [reply ml_api/server.py]: updated error code to 422.

**Comment by kennethjiang** [comment]
Thank you @nvtkaszpir

**Pull request merged**